### ISSUE TITLE
Servlet API 3.0 multipart support does not work for content-type `application/json`

### ISSUE THREAD

**Issue by ashu-walmart** [graphql-java/graphql-java-servlet#84]
@nrouge, @ieugen I have been trying to test the changes for multipart in the last commit - 387ac86. I am testing with Tomcat and a request with content-type as `application/json`. I cant get the request.getParts() to work, I get InvalidContentTypeException.

```
javax.servlet.ServletException: org.apache.tomcat.util.http.fileupload.FileUploadBase$InvalidContentTypeException: the request doesn't contain a multipart/form-data or multipart/mixed stream, content type header is application/json;charset=UTF-8
	at org.apache.catalina.connector.Request.parseParts(Request.java:2909) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at org.apache.catalina.connector.Request.getParts(Request.java:2777) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at org.apache.catalina.connector.RequestFacade.getParts(RequestFacade.java:1084) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at graphql.servlet.GraphQLServlet.lambda$new$2(GraphQLServlet.java:129) ~[graphql-java-servlet-5.0.2.jar:na]
	at graphql.servlet.GraphQLServlet.doRequest(GraphQLServlet.java:261) ~[graphql-java-servlet-5.0.2.jar:na]
	at graphql.servlet.GraphQLServlet.doPost(GraphQLServlet.java:279) ~[graphql-java-servlet-5.0.2.jar:na]
```

This is after the allowCasualMultipartParsing in Catalina context is enabled. Without that setting the exception is:
```
java.lang.IllegalStateException: Unable to process parts as no multi-part configuration has been provided
	at org.apache.catalina.connector.Request.parseParts(Request.java:2811) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at org.apache.catalina.connector.Request.getParts(Request.java:2777) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at org.apache.catalina.connector.RequestFacade.getParts(RequestFacade.java:1084) ~[tomcat-embed-core-8.5.28.jar:8.5.28]
	at graphql.servlet.GraphQLServlet.lambda$new$2(GraphQLServlet.java:129) ~[graphql-java-servlet-5.0.2.jar:na]
	at graphql.servlet.GraphQLServlet.doRequest(GraphQLServlet.java:261) ~[graphql-java-servlet-5.0.2.jar:na]
	at graphql.servlet.GraphQLServlet.doPost(GraphQLServlet.java:279) ~[graphql-java-servlet-5.0.2.jar:na]
```

**Comment by ashu-walmart** [comment]
A potential solution is to rely on content-type instead of `request.getParts()` for identifying a post request to be handled as multipart.

**Comment by nrouge** [comment]
Sorry for that, I made another pull request to fix this issue.

**Comment by oliemansm** [comment]
Fixed in commit [062cdce](https://github.com/graphql-java/graphql-java-servlet/commit/062cdce55a44c4dabd41c112e701b10f9cc79a04)
